File: 7KYU/Will you survive the zombie onslaught?/solution.js
function zombie_shootout(zombies, range, ammo) {
    console.log("zombies: " + zombies);
    console.log("range: " + range);
    console.log("ammo: " + ammo);
    var zombiesThatCanBeShot = (range * 2);
    if(zombiesThatCanBeShot > ammo) {
      if(ammo < zombies) {
        return `You shot ${ammo} zombies before being eaten: ran out of ammo.`;
      }
    } else {
      if(zombiesThatCanBeShot < zombies)
        return `You shot ${zombiesThatCanBeShot} zombies before being eaten: overwhelmed.` 
    }
    return `You shot all ${zombies} zombies.`;
}

File: 7KYU/Will you survive the zombie onslaught?/solution.go
package kata

import (
	"fmt"
)

func Zombie_shootout(zombies, initial_range, ammo int) string {

	zombies_that_can_be_shot := initial_range * 2

	if zombies_that_can_be_shot > ammo {
		if ammo < zombies {
			return fmt.Sprintf("You shot %d zombies before being eaten: ran out of ammo.", ammo)
		}
	} else {
		if zombies_that_can_be_shot < zombies {
			return fmt.Sprintf("You shot %d zombies before being eaten: overwhelmed.", zombies_that_can_be_shot)
		}
	}
	return fmt.Sprintf("You shot all %d zombies.", zombies)
}
